x-pack/plugins/security_solution/public/detections/components/rules/data_view_selector/index.tsx

```diff
@@ -53,15 +54,30 @@ export const DataViewSelector = ({ kibanaDataViews, field }: DataViewSelectorPro
       : []
   );
 
+  const [showDataViewAlertsOnAlertsWarning, setShowDataViewAlertsOnAlertsWarning] = useState(false);
+
   useEffect(() => {
     if (!selectedDataViewNotFound && dataViewId) {
-      setSelectedOption([
-        { id: kibanaDataViews[dataViewId].id, label: kibanaDataViews[dataViewId].title },
-      ]);
+      const dataViewsTitle = kibanaDataViews[dataViewId].title;
+      const dataViewsId = kibanaDataViews[dataViewId].id;
+
+      if (dataViewsId === 'security-solution-default') {
+        setShowDataViewAlertsOnAlertsWarning(true);
+      } else {
+        setShowDataViewAlertsOnAlertsWarning(false);
+      }
```

Comment:
nit: could possibly condense this into

```suggestion
      const dataViewsTitle = kibanaDataViews[dataViewId].title;  
  setShowDataViewAlertsOnAlertsWarning(kibanaDataViews[dataViewId].id === 'security-solution-default');

    
```